homeassistant/components/plugwise/binary_sensor.py

```diff
@@ -92,25 +85,18 @@ class PlugwiseBinarySensorEntity(PlugwiseEntity, BinarySensorEntity):
 
     def __init__(
         self,
-        api: Smile,
         coordinator: PlugwiseDataUpdateCoordinator,
-        name: str,
-        dev_id: str,
+        device_id: str,
         description: BinarySensorEntityDescription,
     ) -> None:
         """Initialise the binary_sensor."""
-        super().__init__(api, coordinator, name, dev_id)
+        super().__init__(coordinator, device_id)
         self.entity_description = description
         self._attr_is_on = False
-        self._attr_unique_id = f"{dev_id}-{description.key}"
-
-        if dev_id == coordinator.data.gateway["heater_id"]:
-            self._entity_name = "Auxiliary"
-
-        self._name = f"{self._entity_name} {description.name}"
-
-        if dev_id == coordinator.data.gateway["gateway_id"]:
-            self._entity_name = f"Smile {self._entity_name}"
+        self._attr_unique_id = f"{device_id}-{description.key}"
+        self._attr_name = (
+            f"{coordinator.data.devices[device_id].get('name')} {description.name}"
```

Comment:
If the name item is missing we'll be stringifying `None` here. Is that wanted?

Reply:
I could make it conditional to include the name. The upstream library doesn't guarantee the name (although it probably does, but has many paths).

Will adjust 👍

Reply:
Adjusted in [bbf3e3b](https://github.com/home-assistant/core/pull/66074/commits/bbf3e3bc737f7e1c35a976f278f681a009e76eef)

Reply:
The upstream library always provides a name, at least that's the intention.
What can we do to improve this?